### ISSUE TITLE
Pod install using cocoapods 1.11.0 fails with a `does not define modules` message

### ISSUE THREAD

**Issue by polmum** [CocoaPods/CocoaPods#10912]
* [X] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install` using the latest cocoapods version (`1.11.0`)

## What did you expect to happen?

Install all pod dependencies correctly, as it was happening with the previous version (`1.10.1`).

## What happened instead?

The command failed with the following message:
```
[!] The following Swift pods cannot yet be integrated as static libraries:

The Swift pod `BidMachine` depends upon `Protobuf`, which does not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.
```

### Stack

```
   CocoaPods : 1.11.0
        Ruby : ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-darwin19]
    RubyGems : 3.0.3
        Host : macOS 11.4 (20F71)
       Xcode : 12.5.1 (12E507)
         Git : git version 2.30.1 (Apple Git-130)
Ruby lib dir : /usr/local/Cellar/ruby@2.6/2.6.6_1/lib
Repositories :  trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/currentuser/.gem/bin/pod
```

### Plugins

```
claide-plugins        : 0.9.2
cocoapods-deintegrate : 1.0.5
cocoapods-keys        : 2.2.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

### Podfile

```ruby
platform :ios, '10.0'

target 'TestApp' do
  pod 'BidMachine'

  target 'TestAppTests' do
    inherit! :search_paths
  end

  target 'TestAppUITests' do

  end
end
```

## Project that demonstrates the issue

[CocoapodsTestApp.zip](https://github.com/CocoaPods/CocoaPods/files/7101456/CocoapodsTestApp.zip)
If you run the sample project with older cocoapods versions, the `pod install` command will succeed.

**Comment by dnkoutso** [comment]
thanks for this great report and sample app. will investigate for 1.11.1

**Comment by dnkoutso** [comment]
while i have not inspected this at all, there is a tiny chance cocoapods 1.11 now catches this case more correctly than previous versions but again I will need to investigate.

**Comment by polmum** [comment]
Yes, I was thinking this might be the case, but I wanted to make sure by letting you know. If this is indeed the case, there will most likely be more users with issues like the one I'm reporting.

**Comment by dnkoutso** [comment]
if it is the case then the error message describes what needs to be done but lets cross that bridge if its true

**Comment by polmum** [comment]
Agreed.
I really hope that's not the case though, as we'd rather not use `use_modular_headers!` globally, and the `:modular_headers => true` flag in this case would have to be used on a indirect dependency (we do not depend upon `Protobuf` directly).

**Comment by dnkoutso** [comment]
a quick look shows that no changes were made related to throwing this warning...........this is kinda weird.

**Comment by dnkoutso** [comment]
@polmum I think I found the culprit and might be "by design" after all.

The pod BidMachine (https://github.com/appodeal/CocoaPods/blob/master/BidMachine/1.7.3.5/BidMachine.podspec) has no sources and its a treated as a vendored pre-compiled pod.

There is a [change](https://github.com/CocoaPods/CocoaPods/pull/10108) in 1.11.0 that inspects whether this pod uses swift or not to fix [issue](https://github.com/CocoaPods/CocoaPods/issues/8649) which now checks if the pod author has specified `swift_versions` in their podspec as the supported version of Swift for a pre-compiled pod. 

So CocoaPods now treats BidMachine as if it uses swift and performs the checks needed. This now causes the inspect to scan BidMachine to ensure its dependencies for static libraries are also using modular headers for its dependencies so the Protobuf pod breaks now.

This is only limited to a very specific scenario or using static libraries and consuming pre-compiled pods.

There is an enhancement somewhere here that asks to skip some dependencies if they are not imported from modular headers but that is for non-vendored frameworks.

Based on this change and fix I think you'd have to `modular_headers` true.

**Comment by dnkoutso** [comment]
Was able to `pod install` and compile successfully with:

```ruby
platform :ios, '10.0'

target 'TestApp' do
  pod 'BidMachine'
  pod 'Protobuf', :modular_headers => true

  target 'TestAppTests' do
    inherit! :search_paths
  end

  target 'TestAppUITests' do

  end
end
```

**Comment by dnkoutso** [comment]
We _could_ maybe update the check to skip targets that do not require building?

**Comment by dnkoutso** [comment]
@polmum I will be updating to relax the validation to only non-prebuilt pods.

**Comment by dnkoutso** [comment]
PR up https://github.com/CocoaPods/CocoaPods/issues/10912

**Comment by polmum** [comment]
Great news! Thanks for your quick replies and for considering relaxing the validation.
I was not really a big fan of adding Protobuf to the podfile, as it was not really a dependency of our project. Also, unless i'm mistaken, an update on the depended version of Protobuf by BidMachine would have forced us to do a `pod update Protobuf` instead of a `pod install`

**Comment by dnkoutso** [comment]
Merged. I will wait a few days for 1.11.1 for new issues to pop up. You can use Bundler in the meantime to point to the `1-11-stable` branch if you need this fix today.

**Comment by umang-simform** [comment]
It worked, thank you :)

**Comment by dnkoutso** [comment]
awesome!

**Comment by dnkoutso** [comment]
I accidentally merged the fix to master, this will now go to 1.11.2.
